### PR TITLE
Add ability to Ignore Classes from ArrayDecorator

### PR DESCRIPTION
**Pull request by yemenifree**
Some custom class have built-in presenter so cause decorator multi
times, for example, Metable extensions of Sofa/Eloquence

### CODE REVIEW

File: src/Decorators/ArrayDecorator.php [hunk 1]
```diff
@@ -45,7 +45,25 @@ public function __construct(AutoPresenter $autoPresenter)
...
+    /**
+     * Ignore a class from decorate.
+     *
+     * @param $subject
+     *
+     * @return bool
+     */
+    public function isNotIgnoreClass($subject)
+    {
+        foreach (config()->get('laravel-auto-presenter.ignore-class-decorate', []) as $class) {
```
Comment (GrahamCampbell) [inline src/Decorators/ArrayDecorator.php]: We can't accept code that looks like this I'm afraid. The config function doesn't exist, so we can't call it. You must resolve the config in the service provider, and pass it in the constructor.

File: src/Decorators/ArrayDecorator.php [hunk 2]
```diff
@@ -23,17 +24,25 @@ class ArrayDecorator implements DecoratorInterface
...
+     *
+     * @var \Illuminate\Contracts\Container\Container
+     */
+    private $container;
 
     /**
      * Create a new array decorator.
      *
      * @param \McCool\LaravelAutoPresenter\AutoPresenter $autoPresenter
+     * @param \Illuminate\Contracts\Container\Container  $container
```
Comment (GrahamCampbell) [inline src/Decorators/ArrayDecorator.php]: I really did mean, please actually resolve the config in the service provider, and just pass the resolved config in here.

Comment (GrahamCampbell) [inline src/Decorators/ArrayDecorator.php]: Don't couple us to the container or the config component.

**Comment by GrahamCampbell** [comment]
Actually, I really don't like the idea of doing this via config. It should be done programmatically.

**Comment by GrahamCampbell** [comment]
I would say, just override the autopresenter binding, and register your own decorators.

**Comment by yemenifree** [comment]
ok override binding good idea

**Pull request closed without merging**